confirm-deposit

```diff
@@ -1 +1,109 @@
-# Title
+# Confirm a Deposit to the Exchange
+
+Deposits are made in several stages. 
+One of the stages is to go through the [Exchange server](https://docs.flare.network/exchange/architecture/#architecture-of-an-exchange), which monitors for deposits and updates the Balances database. 
+Run the code below to have the Exchange server listen for deposit transactions and confirm the deposits.
+
+On a blockchain, multiple [validators](h[ttps://docs.flare.network/tech/glossary/](https://docs.flare.network/tech/validators/)) examine newly created transactions and come to an agreement about whether they are valid. 
+When they are valid, they are added to a new block on the chain and identified by a one-way cryptographic algorithm, called a _hash_. 
+The one-way hash prevents transactions from being reverted. 
+[Blocks](https://docs.flare.network/tech/glossary/) are added to the [blockchain](https://docs.flare.network/tech/glossary/) in chronological order and it may take a few blocks for the hash to be completed.
+To prevent problems, we confirm deposits several blocks back (five blocks in our code), when the blocks are most likely to be irreversible.
+
+Newly submitted deposits to our receiving address are labeled "pending."
+After five blocks have been created, valid deposits are labeled "confirmed."
+
+Run the javascript below to see pending and confirmed transactions and which block number they were added to.
+
+## One-Time Setup
+
+1. Install [node.js](https://nodejs.org/en/download/). 
+On the command line, run `node -v`  to check your installation.
+
+2. Clone the Flare repository: 
+
+`git clone https://github.com/flare-foundation/flare.git
+cd flare`
+
+See [the Flare repository on GitHub](https://github.com/flare-foundation/flare).
+
+3. Install the web3.js library: 
+
+`npm install web3`
+
+See [Adding web3.js](https://web3js.readthedocs.io).
+
+4. Build Flare with this script:
+
+`./scripts/build.sh`
+
+## Run the Code
+
+1. `cd` to `flare`
+
+2. Run the code: 
+
+The following code:
+
+* Sets up the deposit by instantiating `web3`, assigning the endpoint URL, and assigning the `receivingAddress`
+* Subscribes to 'pending transactions` to listen for deposit transactions and, if they are sent to the receiving address, labels them as "pending."
+* Subscribes to 'newBlockHeaders` to get a block that is five blocks back from the block with the pending transactiions and labels it as "confirmed."
+
```

Comment:
Since multiple transactions may be added to a block at a time, are we marking 1 pending transaction at a time, and then the entire set (1 or more) of confirmed transactions in a block?

Reply:
This is exactly what the code does, yes!
The `pendingTransactions` event is received for each individual transaction, whereas the `newBlockHeaders` event retrieves a block, and then iterates over all the transactions contained in it (the `forEach` loop).

Reply:
Of course! Now I see: `getTransaction` and `forEach`!